src/analysis/pluginDeclaration.js

```diff
@@ -23,6 +26,18 @@ export type PluginDeclaration = {|
   +userTypes: $ReadOnlyArray<NodeType>,
 |};
 
+export type PluginDeclarations = $ReadOnlyArray<PluginDeclaration>;
+
+export function toJSON(decs: PluginDeclarations): PluginDeclarationsJSON {
+  return toCompat(COMPAT_INFO, decs);
+}
+
+export function fromJSON(json: PluginDeclarationsJSON): PluginDeclarations {
+  return fromCompat(COMPAT_INFO, json);
+}
+
+export type PluginDeclarationsJSON = Compatible<PluginDeclarations>;
```

Comment:
Nit: I don't think `PluginDeclarationsJSON` needs to be exported.

Reply:
We don't need it yet, but for consistency with our other data formats, I will export it. Since we're writing it to disk, it makes sense that we'd allow others to know about the type.

Reply:
*Weeelll*, as we found out with Projects here https://github.com/sourcecred/sourcecred/pull/1518
The real signature for the fromJSON inputs are `Compatible<any>` (maybe even just `any`).

Because they're just an object we parsed from JSON and still needs runtime validation. With that in mind, I think there's rarely any case where we would use the type. For a return type it makes sense, but in practice it tends to be a one-liner which serializes it immediately after.

```js
stringify(toJSON(thing));
```